### DISCUSSION TITLE
Thermostat current temperature (measured) goes to 0.0

### DISCUSSION THREAD

**Discussion by soyxan** [arendst/Tasmota#22074] [Q&A]
### PROBLEM DESCRIPTION
_A clear and concise description of what the problem is._
Hi, I am testing the Thermostat implementation with an external sensor and I am facing the following issue: My external temperature sensor gets updated only when a temperature change is detected so it could take more than 1 hour between updates and therefore the calls to `TempMeasuredSet` command. I have noticed that some time between updates the reported Current Temperature in the web UI goes to 0.0ºC (affecting the Thermostat action) till a new update comes and a new call is made to `TempMeasuredSet` and the reported Current Temperature goes back to the sensor value:

![image](https://github.com/user-attachments/assets/18354454-7b4d-46c9-b9ac-899fee2b6c7c)

I have checked the rsync logs (attached) and I do not see anything strange. So, my questions are: Do the `TempMeasuredSet` command needs to be called more often to avoid this situation? Is this the expected behaviour?
With the internal temperature sensor (which is updated more often) I do not have this issue.

This is the Berry code I defined to provide the command to update the measured temperature (called when the external temperature sensor is updated):

```
def set_ext_temp2(cmd, idx, payload, payload_json)
    tasmota.set_timer(0, /->tasmota.cmd(string.format("TempMeasuredSet %f", number(payload))))
    tasmota.resp_cmnd_done()
end
tasmota.add_cmd("SetExtTemp2", set_ext_temp2)
```

### REQUESTED INFORMATION
_Make sure your have performed every step and checked the applicable boxes before submitting your issue. Thank you!_

- [x] Read the [Contributing Guide and Policy](https://github.com/arendst/Tasmota/blob/development/CONTRIBUTING.md) and [the Code of Conduct](https://github.com/arendst/Tasmota/blob/development/CODE_OF_CONDUCT.md)
- [x] Searched the problem in [issues](https://github.com/arendst/Tasmota/issues)
- [x] Searched the problem in [discussions](https://github.com/arendst/Tasmota/discussions)
- [x] Searched the problem in the [docs](https://tasmota.github.io/docs/FAQ)
- [x] Searched the problem in the [chat](https://discord.gg/Ks2Kzd4)
- [x] Device used (e.g., Sonoff Basic): Sonoff NsPanel
- [x] Tasmota binary firmware version number used: 12.5.0
  - [ ] Pre-compiled
  - [x] Self-compiled
- [ ] Flashing tools used: _____
- [x] Provide the output of command: `Backlog Template; Module; GPIO 255`:
```lua
  Configuration output here:
13:50:00.385 CMD: Backlog Template; Module; GPIO 255
13:50:00.440 MQT: stat/tasmota_FA6CE8/RESULT = {"NAME":"NSPanel","GPIO":[0,0,0,0,3872,0,0,0,0,0,32,0,0,0,0,225,0,480,224,1,0,0,0,33,0,0,0,0,0,0,0,0,0,0,4736,0],"FLAG":0,"BASE":1}
13:50:00.641 MQT: stat/tasmota_FA6CE8/RESULT = {"Module":{"0":"NSPanel"}}
13:50:00.857 MQT: stat/tasmota_FA6CE8/RESULT = {"GPIO0":{"0":"None"},"GPIO1":{"0":"None"},"GPIO2":{"0":"None"},"GPIO3":{"0":"None"},"GPIO4":{"3872":"Output Lo"},"GPIO5":{"0":"None"},"GPIO6":{"0":"None"},"GPIO7":{"0":"None"},"GPIO8":{"0":"None"},"GPIO9":{"0":"None"},"GPIO10":{"0":"None"},"GPIO11":{"0":"None"},"GPIO12":{"0":"None"},"GPIO13":{"0":"None"},"GPIO14":{"32":"Button1"},"GPIO15":{"0":"None"},"GPIO16":{"0":"None"},"GPIO17":{"0":"None"},"GPIO18":{"0":"None"},"GPIO19":{"225":"Relay2"},"GPIO20":{"0":"None"},"GPIO21":{"480":"Buzzer"},"GPIO22":{"224":"Relay1"},"GPIO23":{"0":"None"},"GPIO24":{"0":"None"},"GPIO25":{"0":"None"},"GPIO26":{"0":"None"},"GPIO27":{"33":"Button2"},"GPIO32":{"0":"None"},"GPIO33":{"0":"None"},"GPIO34":{"0":"None"},"GPIO35":{"0":"None"},"GPIO36":{"0":"None"},"GPIO37":{"0":"None"},"GPIO38":{"4736":"ADC Temp1"},"GPIO39":{"0":"None"}}

```
- [x] If using rules, provide the output of this command: `Backlog Rule1; Rule2; Rule3`:
```lua
  Rules output here:
temp  26.9
13:51:08.236 CMD: Backlog Rule1; Rule2; Rule3
13:51:08.278 MQT: stat/tasmota_FA6CE8/RESULT = {"Rule1":{"State":"OFF","Once":"OFF","StopOnError":"OFF","Length":0,"Free":511,"Rules":""}}
13:51:08.486 MQT: stat/tasmota_FA6CE8/RESULT = {"Rule2":{"State":"OFF","Once":"OFF","StopOnError":"OFF","Length":0,"Free":511,"Rules":""}}
13:51:08.689 MQT: stat/tasmota_FA6CE8/RESULT = {"Rule3":{"State":"OFF","Once":"OFF","StopOnError":"OFF","Length":0,"Free":511,"Rules":""}}

```
- [x] Provide the output of this command: `Status 0`:
```lua
  STATUS 0 output here:


```
- [x] Set `weblog` to 4 and then, when you experience your issue, provide the output of the Console log:
Log attached, I am not able to see when the issue occurred as it is not shown in the logs
[syslog_tasmota.txt](https://github.com/user-attachments/files/16837772/syslog_tasmota.txt)

### TO REPRODUCE
_Steps to reproduce the behavior:_

### EXPECTED BEHAVIOUR
_A clear and concise description of what you expected to happen._

### SCREENSHOTS
_If applicable, add screenshots to help explain your problem._

### ADDITIONAL CONTEXT
_Add any other context about the problem here._

**(Please, remember to close the issue when the problem has been addressed)**

**Comment by sfromis** [comment]
One detail I noticed is that your created command name `SetExtCmd2` cannot work, as command names must not end with a number. This is due to the common structure that a number appended to a command name is an index (passed in the `idx` parameter). However, your log did show that you did not use what you posted, but instead command name `SetExtTemp`.

Did you try doing a test with quicker updates, than very long intervals not good for maintaining a gradient calculation? Using a `cron` timer, you could do something like issuing the command at fixed shorter intervals, and then have the command updating a global variable with last known value.

Not sure how important this is (or if it really matters), but I'd prefer a higher precision than merely 1 decimal, again due to calculations of gradient.
